specs/core/1_shard-data-chains.md

```diff
@@ -122,3 +131,118 @@ Verify that the `shard_block_combined_data_root` is the output of these function
 ### Shard block fork choice rule
 
 The fork choice rule for any shard is LMD GHOST using the validators currently assigned to that shard, but instead of being rooted in the genesis it is rooted in the block referenced in the most recent accepted crosslink (ie. `state.crosslinks[shard].shard_block_hash`). Only blocks whose `beacon_chain_ref` is the block in the main beacon chain at the specified `slot` should be considered (if the beacon chain skips a slot, then the block at that slot is considered to be the block in the beacon chain at the highest slot lower than a slot).
+
+## Changes to beacon chain
+
+Change to attestation verification:
+
+* Let `expected_depth = log2(SHARD_BLOCK_SIZE // SHARD_CHUNK_SIZE * next_power_of_2(slot - last_crosslink_slot))`. For each integer `i = 0, 1`, let `messages[i]` be `AttestationData(slot, shard, parent_hashes, shard_block_hash, expected_depth, attestation_indices.total_validator_count, justified_slot)`.
+
+A `ProofOfCustodyChallenge` has the following fields:
+
+```python
+{
+    # Which validator is responding
+    'responder_index': 'uint64',
+    # Seed hash
+    'seed_hash': 'hash32',
+    # Depth
+    'depth': 'uint64',
+    # Index in tree
+    'data_index': 'uint64',
+    # Expiry date
+    'expiry_slot': 'uint64',
+    # Who is challenging
+    'challenger_index': 'uint64',
+    # Data root
+    'data_root': 'hash32',
+    # Proof of custody bit
+    'bit': 'bool'
+}
+```
+
+
+
+Three new types of `SpecialObject` records:
+
+#### PROOF_OF_CUSTODY_SEED_CHANGE
+
+```python
+{
+    'index': 'uint64',
+    'new_commitment': 'hash32',
+    'signature': ['uint256']
+}
+```
+
+* Let `signed_data` be `bytes8(fork_version) + new_commitment`.
+* Verify that `BLSVerify(pub=state.validators[index].pubkey, msg=hash(signed_data), sig=signature)` passes.
+* Verify that `hash(new_commitment) = state.validators[index].proof_of_custody_commitment`.
+* Verify that `block.slot >= state.validators[index].proof_of_custody_last_change + PROOF_OF_CUSTODY_MIN_CHANGE_PERIOD`.
+* Set `state.validators[index].proof_of_custody_second_last_change` to `state.validators[index].proof_of_custody_last_change`.
+* Set `state.validators[index].proof_of_custody_last_change` to `block.slot`.
+
+#### PROOF_OF_CUSTODY_CHALLENGE
+
+```
+{
+    'attestation': SpecialAttestationData,
+    'validator_index': 'uint64',
+    'data_index': 'uint64',
+}
+```
+
+Perform the following checks:
+
+* Verify that the attestation is valid.
+* Verify that `attestation.slot > state.validators[index].proof_of_custody_second_last_change`.
+* Verify that `validator_index in attestation.aggregate_sig_poc_0_indices` or `validator_index in attestation.aggregate_sig_poc_1_indices`; if the former, let `bit = False`, else `bit = True`.
+* Verify that `state.validators[index].status == ACTIVE` or `state.validators[index].status == PENDING_EXIT` and `block.slot < state.validators[index].exit_slot + PROOF_OF_CUSTODY_MIN_CHANGE_PERIOD`
```

Comment:
Should a validator have to respond to PoC challenges if `EXITED` but not yet `WITHDRAWN`?
I suppose not. An `EXITED` validator shouldn't have to be online, and can only be punished for previously signed slashable messages.